demo/movielens/online/src/main/java/com/dmetasoul/metaspore/demo/movielens/diversify/diversifier/impl/MaximalMarginalRelevanceDiversifier.java

```diff
@@ -54,8 +58,8 @@ public List<ItemModel> diverse(RecommendContext recommendContext,
         // compute the genre in the window
         HashMap<String, Integer> genreInWindow = new HashMap();
         HashMap<String, Integer> genreSplitedInWindow = new HashMap<>();
-
         // start diverse
+        ListNode itemNode = itemLinkList.head.next;
```

Comment:
We should check the `itemLinkList`, `itemLinkList.head` , `itemLinkList.next` are not `null` here.